account_invoice_merge/models/account_invoice_line.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from itertools import groupby
+from operator import attrgetter
+
+from odoo import api, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def _get_sale_fields(self):
+        return [
+            'sale_line_ids',
+        ]
+
+    @api.model
+    def _get_invoice_line_group_fields(self):
+        """Fields that invoice lines are getting grouped upon.
+
+        This are the fields that will be carried over from the old invoice
+        lines to the new one. Content of this fields should be strictly
+        equal for the invoice lines to be merged.
+        """
+        fields = [
+            'name',
+            'origin',
+            'discount',
+            'invoice_line_tax_ids',
+            'price_unit',
+            'product_id',
+            'account_id',
+            'account_analytic_id',
+            'uom_id',
+            'analytic_tag_ids',
```

Comment:
Why not copying `sequence` too?

Reply:
Does it really make sense in the new invoice the line sequence from all other invoices?

Reply:
The more i think about it - no, it depends on the customer using it. I have a customer where this is needed, but with this refactor more parts can be manipulated or extended. That closure in the original code is a pain in the ass.

Reply:
@intero-chz @mpanarin can I merge without this?

Reply:
@elicoidal for me this PR is fine, but i can't do something about the merge itself :baby: